doc: update INSTALL.md about opam and platform versions

Alizter opened the pull request:
I've added some more information about the Coq Platform and opam versions as discussed in #2045.

jdchristensen reviewed (commented):
Looks fine to me, but I haven't tested the different  install methods.

mikeshulman reviewed (commented):
Looks good, thanks.  If the development versions in opam still work, can we add something about how to use them?

Alizter commented:
@mikeshulman I checked myself, and we need the `coq.dev` version of `coq`. This is found in
```
$ opam repo add coq-core-dev https://coq.inria.fr/opam/core-dev
```
so I've added that to `INSTALL.md`. I also added a note about pinning Coq itself, as Coq master may have issues and we don't want constant rebuilds.

Alizter commented:
@mikeshulman Any updates on this?

mikeshulman commented on INSTALL.md:
```diff
@@ -66,12 +79,23 @@ library is `coq-hott` inside the coq-archive.
 
+This will make `coq.dev` the latest available version of `coq`. You can pin
+`coq` to a stable version by running `opam pin add coq.dev 8.19.1` for example.
+
```
```suggestion
Then install the library with `opam install coq-hott`, as for the released version.

```
